### PR TITLE
add a way for mods other mods to change health bars behaviour

### PR DESCRIPTION
**Pull request by olliy1or**

### CODE REVIEW

**Comment by savilli** [comment]
Why?

**Comment by LoneWolfHT** [comment]
> Why?

They are currently working on making the spectator mod functional

**Review by LoneWolfHT — CHANGES_REQUESTED** [review]
You need to add `ctf_hpbar` to the .luacheckrc at line 6

File: mods/ctf/ctf_hpbar/init.lua [hunk 1]
```diff
@@ -50,12 +51,18 @@ minetest.register_playerevent(function(player, eventname)
...
-	entity:set_attach(player, "", {x=0, y=19, z=0}, {x=0, y=0, z=0})
-	players[player:get_player_name()] = {entity=entity}
+    if not ctf_hpbar.can_show(player) then return end
+
+    local entity = minetest.add_entity(player:get_pos(), "ctf_hpbar:entity")
+    entity:set_attach(player, "", {x=0, y=19, z=0}, {x=0, y=0, z=0})
+    players[player:get_player_name()] = {entity=entity}
 
-	update_entity(player)
+		update_entity(player)
```
Comment (LoneWolfHT) [inline mods/ctf/ctf_hpbar/init.lua:65]: If you're using VSCode please open this file and run `Ctrl+Shift+P -> Convert Indentation to Tabs`

**Review by LoneWolfHT — CHANGES_REQUESTED** [review]

**Pull request merged**